internal/server/service/user.go

```diff
@@ -183,6 +184,11 @@ func (s *userServiceBase) password(user *model.User, params UpdatePasswordParams
 	user.Password = pw
 	user.PasswordUpdatedAt = time.Now().Unix()
 
+	// Only update email, when parameter available
```

Comment:
According to your test, `params.NewPassword` can be empty? It will break the statements above?

Reply:
No, `params.NewPassword` can not be empty. The tests use still the same `NewPassword` parameter from the map: https://github.com/mdouchement/standardfile/blob/eff221113dc6053bde569f7ec1be90e8b7a3930d/internal/server/auth_handler_20200115_test.go#L404

Reply:
Ok after some tests, I now remember that both the encryption password and authentication password  are derived from the email and the user's password. So the there is always a new password when the email is changed.